src/scirpy/datasets/__init__.py

```diff
@@ -124,6 +125,34 @@ def maynard2020() -> MuData:
     return mudata.read_h5mu(fname)
 
 
+@_doc_params(
+    processing_code=indent(_read_to_str(HERE / "_processing_scripts/maynard2020.py"), " " * 8),
+    pooch_info=_POOCH_INFO,
+)
+def stephenson2021_5k() -> MuData:
+    """\
+    Return the dataset from :cite:`Maynard2020` as AnnData object.
+
+    21k cells from NSCLC profiled with Smart-seq2, of which 3,500 have :term:`TCRs<TCR>`
+    and 1,500 have :term:`BCRs<BCR>`.
+
+    {pooch_info}
+
+    The raw FASTQ files have been obtained from `PRJNA591860 <https://www.ebi.ac.uk/ena/browser/view/PRJNA591860>`__
+    and processed using the nf-core `RNA-seq pipeline <https://github.com/nf-core/rnaseq>`_ to obtain
+    gene expression and TraCeR/BraCeR to reconstruct receptors.
+
+    The processed files have been imported and transformed into an :class:`anndata.AnnData`
+    object using the following script:
+
+    .. code-block:: python
+
+        {processing_code}
+    """
```

Comment:
@MKanetscheider, could you please update the description here? 

For the other datasets, there are python scripts that specify exactly how the dataset was processed, e.g. https://github.com/scverse/scirpy/blob/main/src/scirpy/datasets/_processing_scripts/maynard2020.py that are included using the `processing_code` reference. 

If you don't have something like that, I'm also happy if you just roughly describe as text how the dataset was processed. Because, let's be honest, likely nobody will ever look at it. 

But at least citation and the reference to where you downloaded the data should be there :)

Reply:
Hi I just wanted to push my changes to this PR, but apparently my access was denied.
![image](https://github.com/user-attachments/assets/9c98a0f6-84b3-4703-8ff4-a902141e8872)
Do you know if that's a problem on my side, or do you have to give some kind of permission? 🥲

Reply:
my bad, you should have received an invite now

Reply:
No problem 😄 
For now, I just added a verbal description, because the code I used is kinda messy as I used to test a lot of different things in the same file... if at any point the code is required I will clean it up and provide here, of course!